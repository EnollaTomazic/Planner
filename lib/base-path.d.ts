export declare function collectPathSegments(value: string | null | undefined): string[];
export declare function normalizeBasePath(value: string | null | undefined): string;
