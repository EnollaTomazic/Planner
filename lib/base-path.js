/**
 * Collect normalized path segments from a string value.
 * @param {string | null | undefined} value
 * @returns {string[]}
 */
export const collectPathSegments = (value) => {
  const trimmed = value?.trim();
  if (!trimmed || trimmed === "/") {
    return [];
  }

  return trimmed
    .split("/")
    .map((segment) => segment.trim())
    .filter((segment) => segment.length > 0);
};

/**
 * Normalize a base path so it can be consumed by Next.js and runtime helpers.
 * @param {string | null | undefined} value
 * @returns {string}
 */
export const normalizeBasePath = (value) => {
  const segments = collectPathSegments(value);
  return segments.length > 0 ? `/${segments.join("/")}` : "";
};
