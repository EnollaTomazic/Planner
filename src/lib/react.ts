import * as React from "react";

export const hasTextContent = (node: React.ReactNode): boolean => {
  if (node === null || node === undefined) return false;
  if (typeof node === "boolean") return false;
  if (typeof node === "string") return node.trim().length > 0;
  if (typeof node === "number") return true;
  if (Array.isArray(node)) return node.some((item) => hasTextContent(item));
  if (React.isValidElement<{ children?: React.ReactNode }>(node)) {
    return hasTextContent(node.props.children ?? null);
  }
  return false;
};

export function observeMediaQuery(
  query: string,
  onChange: (matches: boolean, mediaQuery: MediaQueryList) => void,
): (() => void) | undefined {
  if (typeof window === "undefined" || typeof window.matchMedia !== "function") {
    return undefined;
  }

  let mediaQuery: MediaQueryList;

  try {
    mediaQuery = window.matchMedia(query);
  } catch {
    return undefined;
  }

  const handleChange = (event: MediaQueryListEvent): void => {
    onChange(event.matches, mediaQuery);
  };

  const handleLegacyChange = (): void => {
    onChange(mediaQuery.matches, mediaQuery);
  };

  onChange(mediaQuery.matches, mediaQuery);

  if (typeof mediaQuery.addEventListener === "function") {
    mediaQuery.addEventListener("change", handleChange);
    return () => {
      mediaQuery.removeEventListener("change", handleChange);
    };
  }

  if (typeof mediaQuery.addListener === "function") {
    mediaQuery.addListener(handleLegacyChange);
    return () => {
      mediaQuery.removeListener(handleLegacyChange);
    };
  }

  return undefined;
}

export function useMatchMedia(
  query: string,
  getServerSnapshot?: () => boolean,
): boolean {
  const getMatches = React.useCallback(() => {
    if (typeof window === "undefined") {
      return typeof getServerSnapshot === "function" ? getServerSnapshot() : false;
    }

    if (typeof window.matchMedia !== "function") {
      return false;
    }

    try {
      return window.matchMedia(query).matches;
    } catch {
      return false;
    }
  }, [query, getServerSnapshot]);

  const [matches, setMatches] = React.useState<boolean>(getMatches);

  React.useEffect(() => {
    return observeMediaQuery(query, (nextMatches) => {
      setMatches(nextMatches);
    });
  }, [query]);

  return matches;
}
