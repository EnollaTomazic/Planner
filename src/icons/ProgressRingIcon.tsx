import * as React from "react";
import { spacingTokens, readNumberToken } from "@/lib/tokens";

interface ProgressRingIconProps {
  pct: number; // 0..100
  size?: number;
}

const PROGRESS_DIAMETER_FALLBACK = spacingTokens[7];
const TRACK_INSET_FALLBACK = spacingTokens[2] / 2;
const STROKE_WIDTH_FALLBACK = spacingTokens[0];

export default function ProgressRingIcon({
  pct,
  size,
}: ProgressRingIconProps) {
  const defaultDiameter = React.useMemo(
    () => readNumberToken("--progress-ring-diameter", PROGRESS_DIAMETER_FALLBACK),
    [],
  );
  const strokeWidth = React.useMemo(
    () => readNumberToken("--progress-ring-stroke", STROKE_WIDTH_FALLBACK),
    [],
  );
  const trackInset = React.useMemo(
    () => readNumberToken("--progress-ring-inset", TRACK_INSET_FALLBACK),
    [],
  );
  const resolvedSize = size ?? defaultDiameter;
  const radius = resolvedSize / 2 - trackInset;
  const circumference = 2 * Math.PI * radius;
  const offset = circumference - (pct / 100) * circumference;
  return (
    <svg
      className="h-full w-full rotate-[-90deg]"
      viewBox={`0 0 ${resolvedSize} ${resolvedSize}`}
      aria-hidden="true"
      focusable="false"
    >
      <circle
        cx={resolvedSize / 2}
        cy={resolvedSize / 2}
        r={radius}
        stroke="currentColor"
        strokeWidth={strokeWidth}
        className="text-foreground/20"
        fill="none"
      />
      <circle
        cx={resolvedSize / 2}
        cy={resolvedSize / 2}
        r={radius}
        stroke="currentColor"
        strokeWidth={strokeWidth}
        strokeLinecap="round"
        strokeDasharray={circumference}
        strokeDashoffset={offset}
        className="text-accent shadow-ring motion-safe:animate-pulse [--ring:var(--accent)]"
        fill="none"
      />
    </svg>
  );
}
