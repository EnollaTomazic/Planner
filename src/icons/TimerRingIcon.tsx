import * as React from "react";
import { getRingMetrics, type RingSize } from "@/lib/tokens";
import { cn } from "@/lib/utils";
import RingNoiseDefs from "./RingNoiseDefs";

interface TimerRingIconProps {
  pct: number; // 0..100
  size?: number | RingSize;
}

const DEFAULT_RING_SIZE: RingSize = "l";

export default function TimerRingIcon({
  pct,
  size,
}: TimerRingIconProps) {
  const uniqueId = React.useId();
  const gradientId = `ring-gradient-${uniqueId}`;
  const noiseFilterId = `ring-noise-${uniqueId}`;
  const ringSize = typeof size === "string" ? size : DEFAULT_RING_SIZE;
  const metrics = React.useMemo(() => getRingMetrics(ringSize), [ringSize]);
  const resolvedSize = typeof size === "number" ? size : metrics.diameter;
  const strokeWidth = metrics.stroke;
  const trackInset = metrics.inset;
  const radius = Math.max(resolvedSize / 2 - trackInset, 0);
  const circumference = 2 * Math.PI * radius;
  const offset = circumference - (pct / 100) * circumference;
  const pulse = pct >= 90;
  const accessibleLabel = `Timer ${pct}% complete`;
  return (
    <svg
      className="h-full w-full rotate-[-90deg]"
      viewBox={`0 0 ${resolvedSize} ${resolvedSize}`}
      role="img"
      aria-label={accessibleLabel}
    >
      <defs>
        <RingNoiseDefs id={noiseFilterId} />
        <linearGradient id={gradientId} x1="0%" y1="0%" x2="100%" y2="0%">
          <stop offset="0%" stopColor="hsl(var(--accent))" />
          <stop offset="100%" stopColor="hsl(var(--accent-2))" />
        </linearGradient>
      </defs>
      <circle
        cx={resolvedSize / 2}
        cy={resolvedSize / 2}
        r={radius}
        stroke="hsl(var(--card-hairline))"
        strokeWidth={strokeWidth}
        fill="none"
      />
      <circle
        cx={resolvedSize / 2}
        cy={resolvedSize / 2}
        r={radius}
        stroke={`url(#${gradientId})`}
        strokeWidth={strokeWidth}
        strokeLinecap="round"
        strokeDasharray={circumference}
        strokeDashoffset={offset}
        className={cn(
          "drop-shadow-[0_0_calc(var(--space-3)/2)_hsl(var(--neon-soft))] transition-[stroke-dashoffset] duration-quick ease-linear motion-reduce:transition-none",
          pulse && "motion-safe:animate-pulse motion-reduce:animate-none",
        )}
        filter={`url(#${noiseFilterId})`}
        fill="none"
      />
    </svg>
  );
}
