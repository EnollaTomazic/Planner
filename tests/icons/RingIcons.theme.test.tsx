import React from "react";
import { render, cleanup } from "@testing-library/react";
import { afterEach, describe, expect, it } from "vitest";
import ProgressRingIcon from "@/icons/ProgressRingIcon";
import TimerRingIcon from "@/icons/TimerRingIcon";
import { VARIANTS, applyTheme } from "@/lib/theme";

afterEach(() => {
  cleanup();
  document.documentElement.removeAttribute("style");
  document.documentElement.className = "";
  delete document.documentElement.dataset.themePref;
});

describe.each(VARIANTS)("Ring icons across themes", ({ id: variant }) => {
  it(`renders ring icons for the ${variant} theme`, () => {
    applyTheme({ variant, bg: 0 });

    const { container } = render(
      <div className="flex gap-4">
        <ProgressRingIcon pct={60} size="m" />
        <TimerRingIcon pct={80} size="l" />
      </div>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
