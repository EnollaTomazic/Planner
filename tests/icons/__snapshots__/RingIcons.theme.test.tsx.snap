// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Ring icons across themes > renders ring icons for the aurora theme 1`] = `
<div
  class="flex gap-4"
>
  <svg
    aria-hidden="true"
    class="h-full w-full rotate-[-90deg]"
    focusable="false"
    viewBox="0 0 64 64"
  >
    <defs>
      <filter
        color-interpolation-filters="sRGB"
        filterUnits="objectBoundingBox"
        height="130%"
        id="ring-noise-:r2:"
        width="130%"
        x="-15%"
        y="-15%"
      >
        <feturbulence
          baseFrequency="1.1"
          numOctaves="3"
          result="noise"
          seed="4"
          type="fractalNoise"
        />
        <fecolormatrix
          in="noise"
          result="monoNoise"
          type="saturate"
          values="0"
        />
        <fecomponenttransfer
          in="monoNoise"
          result="noiseAlpha"
        >
          <fefunca
            slope="0.1"
            type="linear"
          />
        </fecomponenttransfer>
        <feblend
          in="SourceGraphic"
          in2="noiseAlpha"
          mode="soft-light"
        />
      </filter>
    </defs>
    <circle
      class="text-foreground/20"
      cx="32"
      cy="32"
      fill="none"
      r="26"
      stroke="currentColor"
      stroke-width="4"
    />
    <circle
      class="text-accent shadow-ring motion-safe:animate-pulse [--ring:var(--accent)]"
      cx="32"
      cy="32"
      fill="none"
      filter="url(#ring-noise-:r2:)"
      r="26"
      stroke="currentColor"
      stroke-dasharray="163.36281798666926"
      stroke-dashoffset="65.34512719466771"
      stroke-linecap="round"
      stroke-width="4"
    />
  </svg>
  <svg
    aria-label="Timer 80% complete"
    class="h-full w-full rotate-[-90deg]"
    role="img"
    viewBox="0 0 224 224"
  >
    <defs>
      <filter
        color-interpolation-filters="sRGB"
        filterUnits="objectBoundingBox"
        height="130%"
        id="ring-noise-:r3:"
        width="130%"
        x="-15%"
        y="-15%"
      >
        <feturbulence
          baseFrequency="1.1"
          numOctaves="3"
          result="noise"
          seed="4"
          type="fractalNoise"
        />
        <fecolormatrix
          in="noise"
          result="monoNoise"
          type="saturate"
          values="0"
        />
        <fecomponenttransfer
          in="monoNoise"
          result="noiseAlpha"
        >
          <fefunca
            slope="0.1"
            type="linear"
          />
        </fecomponenttransfer>
        <feblend
          in="SourceGraphic"
          in2="noiseAlpha"
          mode="soft-light"
        />
      </filter>
      <lineargradient
        id="ring-gradient-:r3:"
        x1="0%"
        x2="100%"
        y1="0%"
        y2="0%"
      >
        <stop
          offset="0%"
          stop-color="hsl(var(--accent))"
        />
        <stop
          offset="100%"
          stop-color="hsl(var(--accent-2))"
        />
      </lineargradient>
    </defs>
    <circle
      cx="112"
      cy="112"
      fill="none"
      r="106"
      stroke="hsl(var(--card-hairline))"
      stroke-width="4"
    />
    <circle
      class="drop-shadow-[0_0_calc(var(--space-3)/2)_hsl(var(--neon-soft))] transition-[stroke-dashoffset] duration-motion-sm ease-linear motion-reduce:transition-none"
      cx="112"
      cy="112"
      fill="none"
      filter="url(#ring-noise-:r3:)"
      r="106"
      stroke="url(#ring-gradient-:r3:)"
      stroke-dasharray="666.0176425610362"
      stroke-dashoffset="133.20352851220719"
      stroke-linecap="round"
      stroke-width="4"
    />
  </svg>
</div>
`;

exports[`Ring icons across themes > renders ring icons for the citrus theme 1`] = `
<div
  class="flex gap-4"
>
  <svg
    aria-hidden="true"
    class="h-full w-full rotate-[-90deg]"
    focusable="false"
    viewBox="0 0 64 64"
  >
    <defs>
      <filter
        color-interpolation-filters="sRGB"
        filterUnits="objectBoundingBox"
        height="130%"
        id="ring-noise-:r8:"
        width="130%"
        x="-15%"
        y="-15%"
      >
        <feturbulence
          baseFrequency="1.1"
          numOctaves="3"
          result="noise"
          seed="4"
          type="fractalNoise"
        />
        <fecolormatrix
          in="noise"
          result="monoNoise"
          type="saturate"
          values="0"
        />
        <fecomponenttransfer
          in="monoNoise"
          result="noiseAlpha"
        >
          <fefunca
            slope="0.1"
            type="linear"
          />
        </fecomponenttransfer>
        <feblend
          in="SourceGraphic"
          in2="noiseAlpha"
          mode="soft-light"
        />
      </filter>
    </defs>
    <circle
      class="text-foreground/20"
      cx="32"
      cy="32"
      fill="none"
      r="26"
      stroke="currentColor"
      stroke-width="4"
    />
    <circle
      class="text-accent shadow-ring motion-safe:animate-pulse [--ring:var(--accent)]"
      cx="32"
      cy="32"
      fill="none"
      filter="url(#ring-noise-:r8:)"
      r="26"
      stroke="currentColor"
      stroke-dasharray="163.36281798666926"
      stroke-dashoffset="65.34512719466771"
      stroke-linecap="round"
      stroke-width="4"
    />
  </svg>
  <svg
    aria-label="Timer 80% complete"
    class="h-full w-full rotate-[-90deg]"
    role="img"
    viewBox="0 0 224 224"
  >
    <defs>
      <filter
        color-interpolation-filters="sRGB"
        filterUnits="objectBoundingBox"
        height="130%"
        id="ring-noise-:r9:"
        width="130%"
        x="-15%"
        y="-15%"
      >
        <feturbulence
          baseFrequency="1.1"
          numOctaves="3"
          result="noise"
          seed="4"
          type="fractalNoise"
        />
        <fecolormatrix
          in="noise"
          result="monoNoise"
          type="saturate"
          values="0"
        />
        <fecomponenttransfer
          in="monoNoise"
          result="noiseAlpha"
        >
          <fefunca
            slope="0.1"
            type="linear"
          />
        </fecomponenttransfer>
        <feblend
          in="SourceGraphic"
          in2="noiseAlpha"
          mode="soft-light"
        />
      </filter>
      <lineargradient
        id="ring-gradient-:r9:"
        x1="0%"
        x2="100%"
        y1="0%"
        y2="0%"
      >
        <stop
          offset="0%"
          stop-color="hsl(var(--accent))"
        />
        <stop
          offset="100%"
          stop-color="hsl(var(--accent-2))"
        />
      </lineargradient>
    </defs>
    <circle
      cx="112"
      cy="112"
      fill="none"
      r="106"
      stroke="hsl(var(--card-hairline))"
      stroke-width="4"
    />
    <circle
      class="drop-shadow-[0_0_calc(var(--space-3)/2)_hsl(var(--neon-soft))] transition-[stroke-dashoffset] duration-motion-sm ease-linear motion-reduce:transition-none"
      cx="112"
      cy="112"
      fill="none"
      filter="url(#ring-noise-:r9:)"
      r="106"
      stroke="url(#ring-gradient-:r9:)"
      stroke-dasharray="666.0176425610362"
      stroke-dashoffset="133.20352851220719"
      stroke-linecap="round"
      stroke-width="4"
    />
  </svg>
</div>
`;

exports[`Ring icons across themes > renders ring icons for the hardstuck theme 1`] = `
<div
  class="flex gap-4"
>
  <svg
    aria-hidden="true"
    class="h-full w-full rotate-[-90deg]"
    focusable="false"
    viewBox="0 0 64 64"
  >
    <defs>
      <filter
        color-interpolation-filters="sRGB"
        filterUnits="objectBoundingBox"
        height="130%"
        id="ring-noise-:rc:"
        width="130%"
        x="-15%"
        y="-15%"
      >
        <feturbulence
          baseFrequency="1.1"
          numOctaves="3"
          result="noise"
          seed="4"
          type="fractalNoise"
        />
        <fecolormatrix
          in="noise"
          result="monoNoise"
          type="saturate"
          values="0"
        />
        <fecomponenttransfer
          in="monoNoise"
          result="noiseAlpha"
        >
          <fefunca
            slope="0.1"
            type="linear"
          />
        </fecomponenttransfer>
        <feblend
          in="SourceGraphic"
          in2="noiseAlpha"
          mode="soft-light"
        />
      </filter>
    </defs>
    <circle
      class="text-foreground/20"
      cx="32"
      cy="32"
      fill="none"
      r="26"
      stroke="currentColor"
      stroke-width="4"
    />
    <circle
      class="text-accent shadow-ring motion-safe:animate-pulse [--ring:var(--accent)]"
      cx="32"
      cy="32"
      fill="none"
      filter="url(#ring-noise-:rc:)"
      r="26"
      stroke="currentColor"
      stroke-dasharray="163.36281798666926"
      stroke-dashoffset="65.34512719466771"
      stroke-linecap="round"
      stroke-width="4"
    />
  </svg>
  <svg
    aria-label="Timer 80% complete"
    class="h-full w-full rotate-[-90deg]"
    role="img"
    viewBox="0 0 224 224"
  >
    <defs>
      <filter
        color-interpolation-filters="sRGB"
        filterUnits="objectBoundingBox"
        height="130%"
        id="ring-noise-:rd:"
        width="130%"
        x="-15%"
        y="-15%"
      >
        <feturbulence
          baseFrequency="1.1"
          numOctaves="3"
          result="noise"
          seed="4"
          type="fractalNoise"
        />
        <fecolormatrix
          in="noise"
          result="monoNoise"
          type="saturate"
          values="0"
        />
        <fecomponenttransfer
          in="monoNoise"
          result="noiseAlpha"
        >
          <fefunca
            slope="0.1"
            type="linear"
          />
        </fecomponenttransfer>
        <feblend
          in="SourceGraphic"
          in2="noiseAlpha"
          mode="soft-light"
        />
      </filter>
      <lineargradient
        id="ring-gradient-:rd:"
        x1="0%"
        x2="100%"
        y1="0%"
        y2="0%"
      >
        <stop
          offset="0%"
          stop-color="hsl(var(--accent))"
        />
        <stop
          offset="100%"
          stop-color="hsl(var(--accent-2))"
        />
      </lineargradient>
    </defs>
    <circle
      cx="112"
      cy="112"
      fill="none"
      r="106"
      stroke="hsl(var(--card-hairline))"
      stroke-width="4"
    />
    <circle
      class="drop-shadow-[0_0_calc(var(--space-3)/2)_hsl(var(--neon-soft))] transition-[stroke-dashoffset] duration-motion-sm ease-linear motion-reduce:transition-none"
      cx="112"
      cy="112"
      fill="none"
      filter="url(#ring-noise-:rd:)"
      r="106"
      stroke="url(#ring-gradient-:rd:)"
      stroke-dasharray="666.0176425610362"
      stroke-dashoffset="133.20352851220719"
      stroke-linecap="round"
      stroke-width="4"
    />
  </svg>
</div>
`;

exports[`Ring icons across themes > renders ring icons for the kitten theme 1`] = `
<div
  class="flex gap-4"
>
  <svg
    aria-hidden="true"
    class="h-full w-full rotate-[-90deg]"
    focusable="false"
    viewBox="0 0 64 64"
  >
    <defs>
      <filter
        color-interpolation-filters="sRGB"
        filterUnits="objectBoundingBox"
        height="130%"
        id="ring-noise-:r4:"
        width="130%"
        x="-15%"
        y="-15%"
      >
        <feturbulence
          baseFrequency="1.1"
          numOctaves="3"
          result="noise"
          seed="4"
          type="fractalNoise"
        />
        <fecolormatrix
          in="noise"
          result="monoNoise"
          type="saturate"
          values="0"
        />
        <fecomponenttransfer
          in="monoNoise"
          result="noiseAlpha"
        >
          <fefunca
            slope="0.1"
            type="linear"
          />
        </fecomponenttransfer>
        <feblend
          in="SourceGraphic"
          in2="noiseAlpha"
          mode="soft-light"
        />
      </filter>
    </defs>
    <circle
      class="text-foreground/20"
      cx="32"
      cy="32"
      fill="none"
      r="26"
      stroke="currentColor"
      stroke-width="4"
    />
    <circle
      class="text-accent shadow-ring motion-safe:animate-pulse [--ring:var(--accent)]"
      cx="32"
      cy="32"
      fill="none"
      filter="url(#ring-noise-:r4:)"
      r="26"
      stroke="currentColor"
      stroke-dasharray="163.36281798666926"
      stroke-dashoffset="65.34512719466771"
      stroke-linecap="round"
      stroke-width="4"
    />
  </svg>
  <svg
    aria-label="Timer 80% complete"
    class="h-full w-full rotate-[-90deg]"
    role="img"
    viewBox="0 0 224 224"
  >
    <defs>
      <filter
        color-interpolation-filters="sRGB"
        filterUnits="objectBoundingBox"
        height="130%"
        id="ring-noise-:r5:"
        width="130%"
        x="-15%"
        y="-15%"
      >
        <feturbulence
          baseFrequency="1.1"
          numOctaves="3"
          result="noise"
          seed="4"
          type="fractalNoise"
        />
        <fecolormatrix
          in="noise"
          result="monoNoise"
          type="saturate"
          values="0"
        />
        <fecomponenttransfer
          in="monoNoise"
          result="noiseAlpha"
        >
          <fefunca
            slope="0.1"
            type="linear"
          />
        </fecomponenttransfer>
        <feblend
          in="SourceGraphic"
          in2="noiseAlpha"
          mode="soft-light"
        />
      </filter>
      <lineargradient
        id="ring-gradient-:r5:"
        x1="0%"
        x2="100%"
        y1="0%"
        y2="0%"
      >
        <stop
          offset="0%"
          stop-color="hsl(var(--accent))"
        />
        <stop
          offset="100%"
          stop-color="hsl(var(--accent-2))"
        />
      </lineargradient>
    </defs>
    <circle
      cx="112"
      cy="112"
      fill="none"
      r="106"
      stroke="hsl(var(--card-hairline))"
      stroke-width="4"
    />
    <circle
      class="drop-shadow-[0_0_calc(var(--space-3)/2)_hsl(var(--neon-soft))] transition-[stroke-dashoffset] duration-motion-sm ease-linear motion-reduce:transition-none"
      cx="112"
      cy="112"
      fill="none"
      filter="url(#ring-noise-:r5:)"
      r="106"
      stroke="url(#ring-gradient-:r5:)"
      stroke-dasharray="666.0176425610362"
      stroke-dashoffset="133.20352851220719"
      stroke-linecap="round"
      stroke-width="4"
    />
  </svg>
</div>
`;

exports[`Ring icons across themes > renders ring icons for the lg theme 1`] = `
<div
  class="flex gap-4"
>
  <svg
    aria-hidden="true"
    class="h-full w-full rotate-[-90deg]"
    focusable="false"
    viewBox="0 0 64 64"
  >
    <defs>
      <filter
        color-interpolation-filters="sRGB"
        filterUnits="objectBoundingBox"
        height="130%"
        id="ring-noise-:r0:"
        width="130%"
        x="-15%"
        y="-15%"
      >
        <feturbulence
          baseFrequency="1.1"
          numOctaves="3"
          result="noise"
          seed="4"
          type="fractalNoise"
        />
        <fecolormatrix
          in="noise"
          result="monoNoise"
          type="saturate"
          values="0"
        />
        <fecomponenttransfer
          in="monoNoise"
          result="noiseAlpha"
        >
          <fefunca
            slope="0.1"
            type="linear"
          />
        </fecomponenttransfer>
        <feblend
          in="SourceGraphic"
          in2="noiseAlpha"
          mode="soft-light"
        />
      </filter>
    </defs>
    <circle
      class="text-foreground/20"
      cx="32"
      cy="32"
      fill="none"
      r="26"
      stroke="currentColor"
      stroke-width="4"
    />
    <circle
      class="text-accent shadow-ring motion-safe:animate-pulse [--ring:var(--accent)]"
      cx="32"
      cy="32"
      fill="none"
      filter="url(#ring-noise-:r0:)"
      r="26"
      stroke="currentColor"
      stroke-dasharray="163.36281798666926"
      stroke-dashoffset="65.34512719466771"
      stroke-linecap="round"
      stroke-width="4"
    />
  </svg>
  <svg
    aria-label="Timer 80% complete"
    class="h-full w-full rotate-[-90deg]"
    role="img"
    viewBox="0 0 224 224"
  >
    <defs>
      <filter
        color-interpolation-filters="sRGB"
        filterUnits="objectBoundingBox"
        height="130%"
        id="ring-noise-:r1:"
        width="130%"
        x="-15%"
        y="-15%"
      >
        <feturbulence
          baseFrequency="1.1"
          numOctaves="3"
          result="noise"
          seed="4"
          type="fractalNoise"
        />
        <fecolormatrix
          in="noise"
          result="monoNoise"
          type="saturate"
          values="0"
        />
        <fecomponenttransfer
          in="monoNoise"
          result="noiseAlpha"
        >
          <fefunca
            slope="0.1"
            type="linear"
          />
        </fecomponenttransfer>
        <feblend
          in="SourceGraphic"
          in2="noiseAlpha"
          mode="soft-light"
        />
      </filter>
      <lineargradient
        id="ring-gradient-:r1:"
        x1="0%"
        x2="100%"
        y1="0%"
        y2="0%"
      >
        <stop
          offset="0%"
          stop-color="hsl(var(--accent))"
        />
        <stop
          offset="100%"
          stop-color="hsl(var(--accent-2))"
        />
      </lineargradient>
    </defs>
    <circle
      cx="112"
      cy="112"
      fill="none"
      r="106"
      stroke="hsl(var(--card-hairline))"
      stroke-width="4"
    />
    <circle
      class="drop-shadow-[0_0_calc(var(--space-3)/2)_hsl(var(--neon-soft))] transition-[stroke-dashoffset] duration-motion-sm ease-linear motion-reduce:transition-none"
      cx="112"
      cy="112"
      fill="none"
      filter="url(#ring-noise-:r1:)"
      r="106"
      stroke="url(#ring-gradient-:r1:)"
      stroke-dasharray="666.0176425610362"
      stroke-dashoffset="133.20352851220719"
      stroke-linecap="round"
      stroke-width="4"
    />
  </svg>
</div>
`;

exports[`Ring icons across themes > renders ring icons for the noir theme 1`] = `
<div
  class="flex gap-4"
>
  <svg
    aria-hidden="true"
    class="h-full w-full rotate-[-90deg]"
    focusable="false"
    viewBox="0 0 64 64"
  >
    <defs>
      <filter
        color-interpolation-filters="sRGB"
        filterUnits="objectBoundingBox"
        height="130%"
        id="ring-noise-:ra:"
        width="130%"
        x="-15%"
        y="-15%"
      >
        <feturbulence
          baseFrequency="1.1"
          numOctaves="3"
          result="noise"
          seed="4"
          type="fractalNoise"
        />
        <fecolormatrix
          in="noise"
          result="monoNoise"
          type="saturate"
          values="0"
        />
        <fecomponenttransfer
          in="monoNoise"
          result="noiseAlpha"
        >
          <fefunca
            slope="0.1"
            type="linear"
          />
        </fecomponenttransfer>
        <feblend
          in="SourceGraphic"
          in2="noiseAlpha"
          mode="soft-light"
        />
      </filter>
    </defs>
    <circle
      class="text-foreground/20"
      cx="32"
      cy="32"
      fill="none"
      r="26"
      stroke="currentColor"
      stroke-width="4"
    />
    <circle
      class="text-accent shadow-ring motion-safe:animate-pulse [--ring:var(--accent)]"
      cx="32"
      cy="32"
      fill="none"
      filter="url(#ring-noise-:ra:)"
      r="26"
      stroke="currentColor"
      stroke-dasharray="163.36281798666926"
      stroke-dashoffset="65.34512719466771"
      stroke-linecap="round"
      stroke-width="4"
    />
  </svg>
  <svg
    aria-label="Timer 80% complete"
    class="h-full w-full rotate-[-90deg]"
    role="img"
    viewBox="0 0 224 224"
  >
    <defs>
      <filter
        color-interpolation-filters="sRGB"
        filterUnits="objectBoundingBox"
        height="130%"
        id="ring-noise-:rb:"
        width="130%"
        x="-15%"
        y="-15%"
      >
        <feturbulence
          baseFrequency="1.1"
          numOctaves="3"
          result="noise"
          seed="4"
          type="fractalNoise"
        />
        <fecolormatrix
          in="noise"
          result="monoNoise"
          type="saturate"
          values="0"
        />
        <fecomponenttransfer
          in="monoNoise"
          result="noiseAlpha"
        >
          <fefunca
            slope="0.1"
            type="linear"
          />
        </fecomponenttransfer>
        <feblend
          in="SourceGraphic"
          in2="noiseAlpha"
          mode="soft-light"
        />
      </filter>
      <lineargradient
        id="ring-gradient-:rb:"
        x1="0%"
        x2="100%"
        y1="0%"
        y2="0%"
      >
        <stop
          offset="0%"
          stop-color="hsl(var(--accent))"
        />
        <stop
          offset="100%"
          stop-color="hsl(var(--accent-2))"
        />
      </lineargradient>
    </defs>
    <circle
      cx="112"
      cy="112"
      fill="none"
      r="106"
      stroke="hsl(var(--card-hairline))"
      stroke-width="4"
    />
    <circle
      class="drop-shadow-[0_0_calc(var(--space-3)/2)_hsl(var(--neon-soft))] transition-[stroke-dashoffset] duration-motion-sm ease-linear motion-reduce:transition-none"
      cx="112"
      cy="112"
      fill="none"
      filter="url(#ring-noise-:rb:)"
      r="106"
      stroke="url(#ring-gradient-:rb:)"
      stroke-dasharray="666.0176425610362"
      stroke-dashoffset="133.20352851220719"
      stroke-linecap="round"
      stroke-width="4"
    />
  </svg>
</div>
`;

exports[`Ring icons across themes > renders ring icons for the ocean theme 1`] = `
<div
  class="flex gap-4"
>
  <svg
    aria-hidden="true"
    class="h-full w-full rotate-[-90deg]"
    focusable="false"
    viewBox="0 0 64 64"
  >
    <defs>
      <filter
        color-interpolation-filters="sRGB"
        filterUnits="objectBoundingBox"
        height="130%"
        id="ring-noise-:r6:"
        width="130%"
        x="-15%"
        y="-15%"
      >
        <feturbulence
          baseFrequency="1.1"
          numOctaves="3"
          result="noise"
          seed="4"
          type="fractalNoise"
        />
        <fecolormatrix
          in="noise"
          result="monoNoise"
          type="saturate"
          values="0"
        />
        <fecomponenttransfer
          in="monoNoise"
          result="noiseAlpha"
        >
          <fefunca
            slope="0.1"
            type="linear"
          />
        </fecomponenttransfer>
        <feblend
          in="SourceGraphic"
          in2="noiseAlpha"
          mode="soft-light"
        />
      </filter>
    </defs>
    <circle
      class="text-foreground/20"
      cx="32"
      cy="32"
      fill="none"
      r="26"
      stroke="currentColor"
      stroke-width="4"
    />
    <circle
      class="text-accent shadow-ring motion-safe:animate-pulse [--ring:var(--accent)]"
      cx="32"
      cy="32"
      fill="none"
      filter="url(#ring-noise-:r6:)"
      r="26"
      stroke="currentColor"
      stroke-dasharray="163.36281798666926"
      stroke-dashoffset="65.34512719466771"
      stroke-linecap="round"
      stroke-width="4"
    />
  </svg>
  <svg
    aria-label="Timer 80% complete"
    class="h-full w-full rotate-[-90deg]"
    role="img"
    viewBox="0 0 224 224"
  >
    <defs>
      <filter
        color-interpolation-filters="sRGB"
        filterUnits="objectBoundingBox"
        height="130%"
        id="ring-noise-:r7:"
        width="130%"
        x="-15%"
        y="-15%"
      >
        <feturbulence
          baseFrequency="1.1"
          numOctaves="3"
          result="noise"
          seed="4"
          type="fractalNoise"
        />
        <fecolormatrix
          in="noise"
          result="monoNoise"
          type="saturate"
          values="0"
        />
        <fecomponenttransfer
          in="monoNoise"
          result="noiseAlpha"
        >
          <fefunca
            slope="0.1"
            type="linear"
          />
        </fecomponenttransfer>
        <feblend
          in="SourceGraphic"
          in2="noiseAlpha"
          mode="soft-light"
        />
      </filter>
      <lineargradient
        id="ring-gradient-:r7:"
        x1="0%"
        x2="100%"
        y1="0%"
        y2="0%"
      >
        <stop
          offset="0%"
          stop-color="hsl(var(--accent))"
        />
        <stop
          offset="100%"
          stop-color="hsl(var(--accent-2))"
        />
      </lineargradient>
    </defs>
    <circle
      cx="112"
      cy="112"
      fill="none"
      r="106"
      stroke="hsl(var(--card-hairline))"
      stroke-width="4"
    />
    <circle
      class="drop-shadow-[0_0_calc(var(--space-3)/2)_hsl(var(--neon-soft))] transition-[stroke-dashoffset] duration-motion-sm ease-linear motion-reduce:transition-none"
      cx="112"
      cy="112"
      fill="none"
      filter="url(#ring-noise-:r7:)"
      r="106"
      stroke="url(#ring-gradient-:r7:)"
      stroke-dasharray="666.0176425610362"
      stroke-dashoffset="133.20352851220719"
      stroke-linecap="round"
      stroke-width="4"
    />
  </svg>
</div>
`;
